### PR TITLE
Refactor ducks

### PR DESCRIPTION
**Pull request by ashmaroli**

### CODE REVIEW

File: src/ducks/metadata.js [hunk 1]
```diff
@@ -62,20 +57,9 @@ export const convertField = (nameAttr, convertType) => ({
...
-export const updateBody = body => ({
-  type: UPDATE_BODY,
-  body,
-});
-
-export const updatePath = path => ({
-  type: UPDATE_PATH,
-  path,
-});
+export const updateTitle = title => ({ type: UPDATE_TITLE, title });
```
Comment (mertkahyaoglu) [inline src/ducks/metadata.js]: One-line is not always good. I had the same feeling years ago but today I find them unnecessary. They are not only hard to read but also hard to make modifications. For example, if we need to do a calculation inside this function, we have to add paranthesis and return statement at the end. Another think is we don't have any concern on too many LOCs or big filesizes. All of the code is minified at the end.

File: src/ducks/config.js [hunk 2]
```diff
@@ -42,9 +42,7 @@ export const putConfig = (config, source = 'editor') => (
 
   // handle errors
   const errors = validateConfig(config);
-  if (errors.length) {
-    return dispatch(validationError(errors));
-  }
+  if (errors.length) return dispatch(validationError(errors));
```
[the remark this answers is not in the document]
Comment (ashmaroli) [reply src/ducks/config.js]: okay..

File: src/ducks/metadata.js (hunk 1 above)
Comment (ashmaroli) [reply src/ducks/metadata.js]: okay..

Comment (ashmaroli) [reply src/ducks/metadata.js]: > All of the code is minified at the end.

True. But the sourcemap isn't minified. That's what increases the size of the published gem.

Comment (mertkahyaoglu) [reply src/ducks/metadata.js]: Hmm. We shouldn't need sourcemap for production though.

Comment (ashmaroli) [reply src/ducks/metadata.js]: That is up for debate. However, the following might help:
https://css-tricks.com/should-i-use-source-maps-in-production/

File: src/ducks/collections.js [hunk 3]
```diff
@@ -145,34 +126,40 @@ export const putDocument = (collection, directory, filename) => (
...
-    }
-    return `${date}-${slugify(metadata.title)}.md`;
+    const docDate = date ? date.split(' ')[0] : moment().format('YYYY-MM-DD');
+    return `${docDate}-${slugifiedTitle}.md`;
   }
-  return `${slugify(metadata.title)}.md`;
+  return `${slugifiedTitle}.md`;
+};
+
+const validateMetadata = (metadata, collection, directory) => {
```
Comment (mertkahyaoglu) [inline src/ducks/collections.js]: This is not just validating metadata but also mutating `path`. I don't think we should mix things here. In order to understand what this function does to path, we need to look into it. It seems not write to me.

Comment (ashmaroli) [reply src/ducks/collections.js]: So, what would be the best way to proceed yet avoid repeating the following logic in every duck (that handles `metadata`)?
```js
const validateMetadata = (metadata, directory) => {
  let { path, title } = metadata;
  let errors = [];

  if ((!path || `${path}/` === directory) && title) {
    path = getFilenameFromTitle(title);
  } else {
    errors = validatePage(metadata);
  }
  return { path, errors };
};
```

**Comment by mertkahyaoglu** [comment]
@ashmaroli Is this ready to go?

**Pull request merged**